library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java

```diff
@@ -907,14 +907,15 @@ private SqlSelection getWhereClause(final Uri uri, final String where, final Str
             selection.appendClause(Downloads.Impl._ID + " = ?", getDownloadIdFromUri(uri));
         }
         if (uriMatch == BATCHES_ID) {
-            selection.appendClause(Downloads.Impl.Batches._ID + " = ?", ContentUris.parseId(uri));
+            selection.appendClause(Downloads.Impl.Batches._ID + " = ?", String.valueOf(ContentUris.parseId(uri)));
```

Comment:
@Electryc yes, you are correct for the -1 and for the null as uri matches "batches/#", hence, getLastPathSegement is the way to go. 

Furthermore, appendClause does a "toString" on the arguments. Maybe, we want to change the signature to String  instead of <T>?